File: app/src/main/java/amrabed/android/release/evaluation/data/converters/ActiveDaysConverter.java
package amrabed.android.release.evaluation.data.converters;

import androidx.room.TypeConverter;

public class ActiveDaysConverter {
    @TypeConverter
    public byte getActiveDays(boolean [] activeDays) {
        byte result = 0;
        for (int i = 0; i < activeDays.length; i++)
        {
            if (activeDays[i])
            {
                result |= (1 << i);
            }

        }
        return result;
    }

    @TypeConverter
    public boolean[] setActiveDays(byte activeDays) {
        boolean [] activeDaysArray = new boolean[7];
        for (int i = 0; i < activeDaysArray.length; i++)
        {
            activeDaysArray[i] = (activeDays & 0x01) == 0x01;
            activeDays >>>= 1;
        }
        return activeDaysArray;
    }
}

File: app/src/main/java/amrabed/android/release/evaluation/data/converters/ActiveDaysConverter.kt
package amrabed.android.release.evaluation.data.converters

import androidx.room.TypeConverter

class ActiveDaysConverter {
    @TypeConverter
    fun getActiveDays(activeDays: BooleanArray): Byte {
        var result = 0
        for (i in activeDays.indices) {
            if (activeDays[i]) {
                result = result or (1 shl i)
            }
        }
        return result.toByte()
    }

    @TypeConverter
    fun setActiveDays(activeDays: Byte): BooleanArray {
        var days :Int = activeDays.toInt()
        val activeDaysArray = BooleanArray(7)
        for (i in activeDaysArray.indices) {
            activeDaysArray[i] = days and 0x01 == 0x01
            days = days ushr 1
        }
        return activeDaysArray
    }
}